dolphinscheduler-task-plugin/dolphinscheduler-task-shell/src/main/java/org/apache/dolphinscheduler/plugin/task/shell/ShellTask.java

```diff
@@ -161,6 +162,19 @@ public AbstractParameters getParameters() {
     private String parseScript(String script) {
         // combining local and global parameters
         Map<String, Property> paramsMap = taskExecutionContext.getPrepareParamsMap();
-        return ParameterUtils.convertParameterPlaceholders(script, ParamUtils.convert(paramsMap));
+        Map<String, String> paramUtilsMap = ParamUtils.convert(paramsMap);
+        return ParameterUtils.convertParameterPlaceholders(script, returnShellTaskPlaceHolders(paramUtilsMap));
+    }
+
+    private Map<String, String> returnShellTaskPlaceHolders(Map<String, String> paramUtilsMap) {
+        Map<String, String> map = new HashMap<>();
+        if(paramUtilsMap != null) {
+            map.putAll(paramUtilsMap);
+        }
+        map.put("WORKFLOW_NAME", taskExecutionContext.getProcessDefinitionName());
```

Comment:
If it is a built-in variable, I think it should be the same for all task types (not only shell). 
And I'm not sure about the benefits of these built-in variables, could you please give some examples?
Furthermore, relevant docs and UT should be added.

Reply:
I have found useful for my use case. We are integrating the flows with the Splunk logs. While logging for the splunk events we are mapping this variable so that we can filter the logs in splunk at the workflow instance level.

Hence, we are using this. Reason we are pushing to splunk is we are not giving the dolphin UI to Users, instead we are giving a utility to run/schedule the jobs in the scheduler.

Reply:
@baratamavinash225 The built-in parameters should be valid for all tasks, so I suggest you start a discussion on the mailing list.